src/main/java/io/fabric8/maven/docker/log/LogDispatcher.java

```diff
@@ -37,15 +40,16 @@ public LogDispatcher(DockerAccess dockerAccess) {
     }
 
     public synchronized void trackContainerLog(String containerId, LogOutputSpec spec) throws FileNotFoundException {
-        LogGetHandle handle = dockerAccess.getLogAsync(containerId, new DefaultLogCallback(spec));
+        LogGetHandle handle = dockerAccess.getLogAsync(containerId, createLogCallback(spec));
         logHandles.put(containerId, handle);
     }
 
     public synchronized void fetchContainerLog(String containerId, LogOutputSpec spec) throws FileNotFoundException {
-        dockerAccess.getLogSync(containerId, new DefaultLogCallback(spec));
+        dockerAccess.getLogSync(containerId, createLogCallback(spec));
     }
 
     public synchronized void untrackAllContainerLogs() {
+        closeLogs();
```

Comment:
This will close all streams from all LogDispatcher instances running. Not sure why it could be more than one, but at least we should avoid the mismatch with the LogCallbackFactory storing the stream in a static Map vs. using object instances for the LogDispatcher here. I would prefer to have LogCallbackFactory to be an object, too.

What's about the following idea: We simply allow `DefaultLogCallback` to reuse an existing printstream (i.e. we do the caching on a static field in DefaultLogCallback along with a usage count) and add a `close()` method to the callback so that the user can signal that he is done with logging ? In `close()` an the associated printstream's usage count will be decreased and if 0 then closed.

I think this kind of resource management makes more sense, also for potential future use cases. If you don't have the cycles to implement this, no problem, I could take over (as this is indeed an important fix to have). Also, a unit tests for `DefaultLogCallback` would be awesome then.